getpaid/backends/pagseguro/__init__.py

```diff
@@ -0,0 +1,183 @@
+        'address_state': 'Estado',  # Estado do cliente (SP) no formato duas letras
+        'phone': 'TelefoneFixo',  # Telefone fixo do cliente
+        'cliente_email': 'Email',  # E-mail do cliente
+    }
+
+    _ACOUNT_DATA = {
+        "email": "",
+        "token":""
+    }
+
+    _PRODUCT_DATA = {
+        "currency": "BRL",
+        "itemId1":"",
+        "itemDescription1":"",
+        "itemQuantity1":"1",
+        "itemAmount1":"",
+        "reference":"",
+    }
+
+
+    def get_gateway_url(self, request):
+        """  """
+        # To active the pagseguro sandbox, will be need run the pagseguro server
+        import uuid
+        is_dev = PaymentProcessor.get_backend_setting('testing', False)
+        if is_dev:
+            gateway_url = "https://ws.sandbox.pagseguro.uol.com.br/v2/"
+            gateway_url2 = "https://sandbox.pagseguro.uol.com.br/v2/"
+        else:
+            gateway_url = "https://ws.pagseguro.uol.com.br/v2/"
+            gateway_url2 = "https://pagseguro.uol.com.br/v2/"
+
+        order = self.payment.order
+
+        token = PaymentProcessor.get_backend_setting('token')
+        email = PaymentProcessor.get_backend_setting('email')
+        self._ACOUNT_DATA["token"] = token
+        self._ACOUNT_DATA["email"] = email
+
+        reference = uuid.uuid4().hex
+
+        product_description = self.get_order_description(self.payment, self.payment.order)
+        redirectURL = PaymentProcessor._get_view_full_url(request, 'getpaid-pagseguro-success', args=(self.payment.id,))
+
+        self._PRODUCT_DATA.update(itemId1=self.payment.id,
+                                  itemDescription1= product_description.encode("latin1", "ignore"), 
+                                  currency=self.payment.currency,
+                                  itemAmount1=self.payment.amount,
+                                  reference=reference,
+                                  redirectURL=redirectURL,
+                                  )
+
+        full_data = dict(self._ACOUNT_DATA, **self._PRODUCT_DATA)
+
+        params = urllib.urlencode(full_data)
+
+        
+        payment_full_url = "%s%s%s" % (gateway_url, self._CHECKOUT_INSTRUCTION_PAGE , params)
+        request.encoding = 'ISO-8859-1'
+        
+        dados = {}
+
+        logger.info(payment_full_url)
+
+        headers = {"Content-type": "application/x-www-form-urlencoded"}
+        response = requests.post(payment_full_url, headers=headers).text
+        
+        code = ""
+        logger.warning("xml response from pagseguro: "+str(response))
+        dom = parseString(response)
+        
+        checkout = dom.getElementsByTagName("checkout")
+        if checkout:
+            childNodes = checkout[0].childNodes
+            code = childNodes[0].firstChild.nodeValue
+            date = childNodes[1].firstChild.nodeValue
+
+        self.payment.external_id = reference
+        self.payment.description = code
```

Comment:
`code` is undefined? 
This backend worke in two steps. The first step we send client data to backend, them we  receive a auth code in the response. This auth code will be used to get payment status when a notification of payment be sent to the site.